src/main/java/xyz/nucleoid/mineout/game/MineoutActive.java

```diff
@@ -286,22 +290,47 @@ private void tickClosing(long time) {
         }
     }
 
+    private static String ordinal(int i) {
+        String[] suffixes = new String[] { "th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th" };
+        switch (i % 100) {
+            case 11:
+            case 12:
+            case 13:
+                return i + "th";
+            default:
+                return i + suffixes[i % 10];
+        }
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
     private void broadcastFinish() {
-        Text message;
+        PlayerSet players = this.gameSpace.getPlayers();
         if (!this.finishRecords.isEmpty()) {
             FinishRecord winningRecord = this.finishRecords.get(0);
 
-            message = new LiteralText("")
-                    .append(new LiteralText(winningRecord.player.getName()).formatted(Formatting.AQUA))
-                    .append(" won the game!")
-                    .formatted(Formatting.GOLD);
+            Text message = new LiteralText("").append(winningRecord.player.getName()).append(" was 1st!").formatted(Formatting.GOLD);
+            players.forEach(player -> {
+                if(players.stream().count() == 1) return;
```

Comment:
PlayerSet can request size without a stream- but it's a bit weird to have this return inside the foreach. Also unimportant mostly stylistic thing, but can use a normal foreach here.

Reply:
Fixing in update